packages/gatsby/src/schema/node-model.js

```diff
@@ -253,6 +413,60 @@ class LocalNodeModel {
   }
 }
 
+class ContextualNodeModel {
```

Comment:
Any reason why we don't use inheritance?

Reply:
I just thought this is cleaner.

Reply:
from my perspective it isn't but I don't have a strong opinion about it.